### ISSUE TITLE
Freeze on login screen after wake from hibernation. 

### ISSUE THREAD

**Issue by vandroiy2013** [acidanthera/bugtracker#386]
This problem occures on Gigabyte GA-Z370M-DS3H. Stuck on login screen aftes wake from hibernation. Hibernatemode=25

**Comment by PMheart** [comment]
Same error here, under macOS 10.14.5 (I cannot even trigger hibernation under 10.15 b1...)

@vandroiy2013 Could you please try this patch? (Which is actually how older version of HibernationFixup works)
10.14.5: `e8 7a 92 00 00 -> 90 90 90 90 90`
10.15 b1: `e8 de 92 00 00 -> 90 90 90 90 90`

**Comment by Andrey1970AppleLife** [comment]
> @vandroiy2013 Could you please try this patch? (Which is actually how older version of HibernationFixup works, so please REMOVE HibernationFixup as well)

1. HibernationFixup perfectly works with OC.
2. For @vandroiy2013 not necessary HibernationFixup, because works without this kext with Clover.

**Comment by PMheart** [comment]
Yeah, I had talked with @lvs1974 days ago too, and he said that the problem might have nothing to do with HBFX itself. I may have to check it from 10.11 to 10.15 this weekend...

Vandroiy's scenario kind of resembled what I got, yet this time I had no luck with Clover either. (It used to work under 10.12 or 10.13 two years ago, and I totally dropped hibernation and switched back to normal sleep)

All in all, time is always something that is hard to find... :(

**Comment by vit9696** [comment]
It seems that on modern macOS systems hibernation got broken on select hardware, and that seems to be bootloader-independent just as @PMheart mentioned. HibernationFixup solves a separate problem, and that is most likely why it does not help. Probably trying with Clover and potentially older macOS versions and AptioMemoryFix versions is most reasonable.

Honestly said, hibernation is a high security risk and Apple no longer supports it for T2 models. Current hibernation support in AptioMemoryFix is based on literal hacks that just happen to work, and we have no idea how to improve them due to completely different memory management compared to Macs. I believe we should rather make it deprecated than attempt to invest into researching for new half-borked heuristics.

**Comment by CobanRamo** [comment]
Hi guys,

Seems to be an ancient fossil problem, I was able to solve this Hibernation 25 (Apple Standby) on every computer I've set up using the following instructions.

Perhaps the whole thing can be transferred to the current OpenCore documentation and Sample.plist, of course after a thorough test on your part.

In the Sample.plist under UEFI/ReservedMemory there is also a Lenovo patch with the same data but the name is somehow not appropriate, this patch in conjunction with a few other things really works on all hackintoshes that I've tested.

Take a closer look at it, maybe we can combine it into a common solution and mark this thread as solved.

Greetings Coban

![HibernationMode25 2](https://github.com/acidanthera/bugtracker/assets/68192365/b8359ae3-6e99-4a97-8e6b-269a02c08c95)
[MacOS HibernationMode EN.pdf](https://github.com/acidanthera/bugtracker/files/15491537/MacOS.HibernationMode.EN.pdf)
[Fix wake from hibernation 25.plist.zip](https://github.com/acidanthera/bugtracker/files/15491540/Fix.wake.from.hibernation.25.plist.zip)
